src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Microsoft.AspNetCore.Razor.Language.csproj

```diff
@@ -2,14 +2,21 @@
 
   <PropertyGroup>
     <Description>Razor is a markup syntax for adding server-side logic to web pages. This package contains the Razor parser and code generation infrastructure.</Description>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <TargetFrameworks>net7.0;netstandard2.0</TargetFrameworks>
```

Comment:
Why did we need the new target framework here?

Reply:
Wait how is this going to work at deployment time? This essentially requires that there are now two copies of MS.ANC.Razor.Language where today there is one copy of MS.ANC.R right? That shouldn't work out unless there are also deployment changes here too.

Reply:
> how is this going to work at deployment time?

I don't see why this would be a problem. One version is for .NET Core, the other for .NET Framework. I guess today .NET Core just reuses the netstandard copy.

> should we just copy the files here with a diff namespace

I don't think it's that easy. It's types like `System.Range` redefined so they can be used in netstandard projects. We can't change their namespace. And we cannot define them in .NET Core.

Reply:
Right, I was thinking about NuGet packages and SDK where there are separate folders for netcore and netstandard DLLs. But I guess you are talking about VS deployment. Not sure how that works, will need to investigate.

Reply:
Btw., here's the problem in an example:

MS.CA.Razor.Workspaces is built today for net7, netstandard, net472 (not sure why all of these). It brings in Razor.Language and Utilities.Shared. With this PR, Razor.Language contains the utilities like `NotNullAttribute`. By removing Workspaces IVT from Utilities, we are fine for netstandard/net472 builds. But net7 build still fails because there's a clash between Razor.Language's netstandard NotNullAttribute and net7 System.Runtime's NotNullAttribute. By making Razor.Language multi-target also net7, it won't include the NotNullAttribute and the build succeeds.

Not sure if it's actually deployed anywhere though. For example, in Visual Studio I see only netstandard DLLs.

<samp>C:\Program Files\Microsoft Visual Studio\2022\Preview\Common7\IDE\CommonExtensions\Microsoft\RazorLanguageServices</samp>

Reply:
> MS.CA.Razor.Workspaces is built today for net7, netstandard, net472 (not sure why all of these)

Because it has to run in multiple worlds.

* net7.0 allows it to run in rzls.exe, a standalone app that supports VS Code.
* net472 because that's what Visual Studio targets.
* netstandard2.0 because we also have a "remote" assembly that runs in the Roslyn OOP process in VS. Because Roslyn OOP can run on either .NET Core or .NET Framework, we have to compile it to netstandard2.0.

Hope that helps!

Reply:
> By removing Workspaces IVT from Utilities, we are fine for netstandard/net472 builds

I am not a huge fan of these sorts of shenanigans. I'm pretty certain they will come back to haunt us later. :smile:

Reply:
> I am not a huge fan of these sorts of shenanigans. I'm pretty certain they will come back to haunt us later. 😄

This is only for servicing branch, where I think some hacks are fine

Reply:
Yeah, I misunderstood that. Jared set me straight yesterday.